thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AnomaliesResource.java

```diff
@@ -784,29 +783,76 @@ private AnomalyDetails constructAnomalyDetails(String metricName, String dataset
     return anomalyDetails;
   }
 
+  // this function generate the time ragne for viewing purpose.
+  private TimeRange getTimeseriesViewingOffsetedTimes(long anomalyStartTime, long anomalyEndTime, DatasetConfigDTO datasetConfig) {
+    TimeUnit dataTimeunit = datasetConfig.getTimeUnit();
+    Period preOffsetPeriod; // The offset we view before the anomaly
+    Period postOffsetPeriod; // The offset we view after the anomaly
+    switch (dataTimeunit) {
+      case DAYS: // 30 days. User needs to see the trend 30 days ahead
+        preOffsetPeriod = new Period(0, 0, 0, 30, 0, 0, 0, 0);
+        postOffsetPeriod = new Period(0, 0, 0, 0, 0, 0, 0, 0);
+        break;
+      case HOURS: // 10 hours
+        preOffsetPeriod = new Period(0, 0, 0, 0, 10, 0, 0, 0);
+        postOffsetPeriod = new Period(0, 0, 0, 0, 10, 0, 0, 0);
+        break;
+      case MINUTES: // 60 minutes
+        preOffsetPeriod = new Period(0, 0, 0, 0, 0, 60, 0, 0);
+        postOffsetPeriod = new Period(0, 0, 0, 0, 0, 60, 0, 0);
+        break;
+      default:
+        preOffsetPeriod = new Period();
+        postOffsetPeriod = new Period();
+    }
+
+    DateTimeZone dateTimeZone = DateTimeZone.forID(datasetConfig.getTimezone());
```

Comment:
This new method and the one below have a lot of repeated code, try and create a reusable snipped from them